modules/openapi-generator/src/main/resources/typescript/types/ObservableAPI.mustache

```diff
@@ -61,19 +62,52 @@ export class Observable{{classname}} {
      * @param {{#required}}{{paramName}}{{/required}}{{^required}}[{{paramName}}]{{/required}}{{#description}} {{description}}{{/description}}
      {{/allParams}}
      */
-    public {{nickname}}WithHttpInfo({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: Configuration): Observable<HttpInfo<{{{returnType}}}{{^returnType}}void{{/returnType}}>> {
-        const requestContextPromise = this.requestFactory.{{nickname}}({{#allParams}}{{paramName}}, {{/allParams}}_options);
+    public {{nickname}}WithHttpInfo({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: Configuration{{^useInversify}}Options{{/useInversify}}): Observable<HttpInfo<{{{returnType}}}{{^returnType}}void{{/returnType}}>> {
+	let _config = this.configuration
+	let allMiddleware: Middleware[] = []
+	{{#useInversify}}
+	if (_options){
+	    _config = _options
+	}
+	allMiddleware = _config?.middleware
+	{{/useInversify}}
+	{{^useInversify}}
+	if (_options && _options.middleware){
+	    // call-time middleware provided
+            let calltimeMiddleware: Middleware[] = _options.middleware
 
+	    switch(_options.middlewareMergeStrategy){
+	    	case 'append':
+                    allMiddleware = this.configuration.middleware.concat(calltimeMiddleware)
+		    break;
+	    	case 'prepend':
+		    allMiddleware = calltimeMiddleware.concat(this.configuration.middleware)
+		    break;
+		default: // replace
```

Comment:
should we set the default explicit ahead of time and throw if we encounter a `never` (unknown value) here at runtime?

Reply:
updated to use default for never case